Source/ARTChannel.m

```diff
@@ -51,6 +51,17 @@ - (void)setOptions:(ARTChannelOptions *)options {
 
 - (void)setOptions_nosync:(ARTChannelOptions *)options {
     _options = options;
+    [self recreateDataEncoderWith:options.cipher];
+}
+
+- (void)recreateDataEncoderWith:(ARTCipherParams*)cipher {
+    NSError *error = nil;
+    _dataEncoder = [[ARTDataEncoder alloc] initWithCipherParams:cipher error:&error];
+    
+    if (error != nil) {
```

Comment:
Should we handle the case where there is an error?

Reply:
it's a copy/paste from init
```objc
- (instancetype)initWithName:(NSString *)name andOptions:(ARTChannelOptions *)options rest:(ARTRestInternal *)rest {
    if (self = [super init]) {
        _name = name;
        _logger = rest.logger;
        _queue = rest.queue;
        _options = options;
        NSError *error = nil;
        _dataEncoder = [[ARTDataEncoder alloc] initWithCipherParams:_options.cipher error:&error];
        if (error != nil) {
            [_logger warn:@"creating ARTDataEncoder: %@", error];
            _dataEncoder = [[ARTDataEncoder alloc] initWithCipherParams:nil error:nil];
        }
    }
    return self;
}
```